tests/unit/customizations/history/test_db.py

```diff
@@ -0,0 +1,711 @@
+        # distinct calls that end with a parsed response.
+        unique_request_ids = set(written_record_request_ids)
+        self.assertEqual(len(unique_request_ids), 3)
+
+        # Check that the request ids match the correct events.
+        first_request_ids = written_record_request_ids[:2]
+        unique_first_request_ids = set(first_request_ids)
+        self.assertEqual(len(unique_first_request_ids), 1)
+
+        second_request_ids = written_record_request_ids[2:4]
+        unique_second_request_ids = set(second_request_ids)
+        self.assertEqual(len(unique_second_request_ids), 1)
+
+        third_request_ids = written_record_request_ids[4:]
+        unique_third_request_ids = set(third_request_ids)
+        self.assertEqual(len(unique_third_request_ids), 1)
+
+    def test_identifier_is_always_the_same(self):
+        # Three http lifecycles are written here, the overall identifier for
+        # the records should remain the same regardless.
+        written_records = self._write_sequence_of_records([
+                {
+                    'event_type': 'api_call',
+                    'payload': '',
+                    'source': 'test',
+                    'timestamp': 1234
+                },
+                {
+                    'event_type': 'parsed_response',
+                    'payload': '',
+                    'source': 'test',
+                    'timestamp': 1234
+                },
+                {
+                    'event_type': 'api_call',
+                    'payload': '',
+                    'source': 'test',
+                    'timestamp': 1234
+                },
+                {
+                    'event_type': 'parsed_response',
+                    'payload': '',
+                    'source': 'test',
+                    'timestamp': 1234
+                },
+                {
+                    'event_type': 'api_call',
+                    'payload': '',
+                    'source': 'test',
+                    'timestamp': 1234
+                },
+                {
+                    'event_type': 'parsed_response',
+                    'payload': '',
+                    'source': 'test',
+                    'timestamp': 1234
+                }
+        ])
+        # The record id is the first field in the record.
+        written_record_ids = [record[0] for record in written_records]
+        # Even though the request lifecycle was reset multpile times the
+        # overall id should be the same.
+        unique_ids = set(written_record_ids)
+        self.assertEqual(len(unique_ids), 1)
+
+
+class ThreadedRecordWriter(object):
+    def __init__(self, writer, fake_connection):
+        self._read_q = queue.Queue()
+        self._write_q = queue.Queue()
+        self._thread = threading.Thread(
+            target=self._threaded_record_writer,
+            args=(writer, fake_connection))
+        self._thread.start()
```

Comment:
I would break this out to its own separate `start()` method. It is a little strange to spin up the actual thread in the initializer.

Reply:
Yea I think also each test for threads may want a different number of threads so I think ill add a `_spin_up_n_threads()` type function that collects them all into a list which is joined on in the `tearDown()`